.ado/release.yml

```diff
@@ -1,239 +1,325 @@
+            condition: succeeded()
+
+          - task: UsePythonVersion@0
+            inputs:
+              versionSpec: $(PYTHON_VERSION)
+            condition: succeeded()
+
+          # Build or restore LLVM
+          - pwsh: New-Item -ItemType Directory -Path $(LLVM_DIRECTORY) -Force
+            displayName: Ensure cache dir exists
+            condition: succeeded()
+
+          - task: Cache@2
+            inputs:
+              key: '"llvm-$(LLVM_VERSION)-${{ target.os }}-${{ target.arch }}"'
+              restoreKeys: '"llvm-$(LLVM_VERSION)-${{ target.os }}-${{ target.arch }}"'
+              path: $(LLVM_DIRECTORY)
+              cacheHitVar: CACHE_RESTORED
+            displayName: Cache LLVM
+            condition: succeeded()
+
+          - pwsh: |
+              Write-Host "##vso[task.setvariable variable=MACOSX_DEPLOYMENT_TARGET;]10.12"
+            displayName: Configure Environment (MACOSX_DEPLOYMENT_TARGET x86_64)
+            condition: eq(variables['Agent.OS'], 'Darwin')
+
+          - pwsh: |
+              git config --global core.longpaths true
+
+              $directory = Resolve-Path $(LLVM_DIRECTORY)
+
+              # Tell the build where LLVM is installed
+              # This is usually automatic, but we are using a different directory than the default
+              Write-Host "##vso[task.setvariable variable=QIRLIB_LLVM_EXTERNAL_DIR;]$directory"
+
+              # Tell the build what llvm feature version to use
+              $feature = "llvm$(LLVM_VERSION)-0"
+              $variableName = "LLVM_SYS_$(LLVM_VERSION)0_PREFIX"
+              Write-Host "##vso[task.setvariable variable=PYQIR_LLVM_FEATURE_VERSION;]$feature"
+
+              # Tell LLVM-SYS where LLVM is installed
+              Write-Host "##vso[task.setvariable variable=$variableName;]$directory"
+
+              # Tell the qirlib build where to cache the LLVM directory and to not download LLVM
+              Write-Host "##vso[task.setvariable variable=QIRLIB_CACHE_DIR;]$directory"
+              Write-Host "##vso[task.setvariable variable=QIRLIB_DOWNLOAD_LLVM;]false"
+            displayName: Configure Environment (ALL)
+            condition: succeeded()
+
+          # prerequisites for building
+          - script: |
+              sudo apt-get install -y ninja-build
+            displayName: Install build dependencies
+            condition: and(succeeded(), eq(variables['Agent.OS'], 'Linux'), ne(variables.CACHE_RESTORED, 'true'))
+
+          - script: |
+              brew install ninja
+            displayName: Install build dependencies
+            condition: and(succeeded(), eq(variables['Agent.OS'], 'Darwin'), ne(variables.CACHE_RESTORED, 'true'))
+
+          - script: |
+              choco install --accept-license -y ninja
+            displayName: Install build dependencies
+            condition: and(succeeded(), eq(variables['Agent.OS'], 'Windows_NT'), ne(variables.CACHE_RESTORED, 'true'))
```

Comment:
It's a shame we have to repeat this every pipeline with the different mechanism for each platform. Being ninja is quite common, is there not an action to install it similar to how there is for Python, Node, Rust, etc.?